src/main/java/xyz/nucleoid/creator_tools/workspace/editor/ServersideWorkspaceEditor.java

```diff
@@ -47,6 +53,47 @@ public ServersideWorkspaceEditor(ServerPlayerEntity player, MapWorkspace workspa
         this.markerEntity = markerEntity;
     }
 
+    @Override
+    public void onEnter() {
+        if (this.canSendPacket(WorkspaceNetworking.WORKSPACE_ENTER_ID)) {
```

Comment:
The intent with the `WorkspaceEditor` type was to use a different implementation for the server-side only case (particles and commands) from the implementation that handles client mods- I think that’s worth doing now, as the implementations will likely diverge more in the future.

Reply:
Would it make sense to require these packets to be handled on an all-or-none basis to fulfill that design? I could alternatively implement per-channel listening using a packet workspace editor and a fallback workspace editor.

Reply:
I think using all-or-none for each player makes a lot of sense! I guess there can be some kind of handshake to establish whether a client knows how to handle the protocol?

Reply:
Sounds good; the `nucleoid_creator_tools:opt_in` packet could potentially contain the mod version and a protocol version number.

Reply:
That sounds like a great approach!

Reply:
Would it make sense to still use `canSendPacket` to avoid creating buffers for packets on channels that aren't registered by the client, or should the `nucleoid_creator_tools:opt_in` packet imply that these packets be handled?

Reply:
Either way works- the latter seems simpler, but maybe there's some gain with regards to versioning/partial protocol support to check specifically

Reply:
Fixed in 3c72c39e504174419ed39a9007d88260f6adbc8f